fedimint-client/src/backup.rs

```diff
@@ -175,14 +179,18 @@ impl Encodable for ClientBackup {
         len += self.metadata.consensus_encode(writer)?;
         len += self.modules.consensus_encode(writer)?;
 
-        // FIXME: this still leaks some information about the backup size if the padding
-        // is so short that its length is encoded as 1 byte instead of 3.
-        let estimated_len = len + 3;
-
-        // Hide small changes in backup size for privacy
-        let alignment_size = Self::get_alignment_size(estimated_len); // +3 for most likely padding len len
-        let padding = vec![0u8; alignment_size - estimated_len];
-        len += padding.consensus_encode(writer)?;
+        // Old-style padding.
+        //
+        // Older version of the client used a custom zero-filled vec to
+        // pad `ClientBackup` to alignment-size here. This has a problem
+        // that the size of encoding of the padding can have different sizes
+        // (due to var-ints).
+        //
+        // Conceptually serialization is also a the wrong place to do padding
+        // anyway, so we moved padding to encrypt/decryption. But for abundance of
+        // caution, we'll keep the old padding around, and always fill it
+        // with an empty Vec.
+        len += Vec::<u8>::new().consensus_encode(writer)?;
```

Comment:
We'd need that if FM supported downgrades, but since we don't it's just abundance of caution.

Reply:
Well, someone might want to recover from backup made by the older version of the app, and then they would have to do full recovery.

And an empty vec is just one `0` byte.

Reply:
Also - this backup is temporarily stored during init of a client with recovery in the database. Very unlikely that someone would upgrade during that exact window, but ... abundance of caution.

Reply:
Oh, I thought partial decode would fix that? But maybe this is wrapped again and partial decode on the outer layer wouldn't help?

Reply:
Old clients expect the vector there (see corresponding `consensus_decode`)

Reply:
It was a stupid way to do padding, but oh well.